### PR TITLE
dbgate: init at 5.2.8

### LINKED ISSUE #306756: Package request: dbgate
**Project description**

Database manager for MySQL, PostgreSQL, SQL Server, MongoDB, SQLite and others. Runs under Windows, Linux, Mac or as web application

**Metadata**

* homepage URL: https://dbgate.org/
* source URL: https://github.com/dbgate/dbgate
* license: mit
* platforms: unix, linux, darwin, windows

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #306756

[Release notes](https://github.com/dbgate/dbgate/releases/tag/v5.2.8)

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by luftmensch-luftmensch** [comment]
```
> nix build .#dbgate
> tree result/
 result
└──  bin
   └──  dbgate
```

**Review by Misterio77 — APPROVED** [review]

File: pkgs/by-name/db/dbgate/package.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    homepage = "https://github.com/dbgate/dbgate";
+    license = licenses.mit;
+    maintainers = with maintainers; [ luftmensch-luftmensch ];
+    changelog = "https://github.com/dbgate/dbgate/blob/master/CHANGELOG.md";
+    mainProgram = "dbgate";
+    platforms = [
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-linux"
+    ];
```
[the remark this answers is not in the document]
Comment (luftmensch-luftmensch) [reply pkgs/by-name/db/dbgate/package.nix:40]: Done!

@Misterio77 Did you also have the capability to merge it?

[the remark this answers is not in the document]
Comment (Misterio77) [reply pkgs/by-name/db/dbgate/package.nix:40]: Sadly no, I might nominate myself for the commit bit at some point tho haha.

**Review by Misterio77 — APPROVED** [review]
LGTM!

**Comment by luftmensch-luftmensch** [comment]
@Misterio77 Since the package had a new release and the PR is still opened I thought it was a good idea to bump the version to 5.2.8 (without the need to make a new pr in a hopefully short time)

The approval should be fast enought

**Comment by Misterio77** [comment]
Result of `nixpkgs-review pr 311405` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dbgate</li>
  </ul>
</details>

**Review by Misterio77 — APPROVED** [review]

File: pkgs/by-name/db/dbgate/package.nix [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        aarch64-linux = {
+          url = "https://github.com/${pname}/${pname}/releases/download/v${version}/${pname}-${version}-linux_arm64.AppImage";
+          hash = "sha256-gxojSSk7prhnd9fy56B9H+Cj6COBLc7xPfV8dTvSO0c=";
+        };
+        x86_64-linux = {
+          url = "https://github.com/${pname}/${pname}/releases/download/v${version}/${pname}-${version}-linux_x86_64.AppImage";
+          hash = "sha256-/Vfd0R+Mzx1CJKkC7dj99pbuuyh8PJtbYlH3wtwVxSM=";
+        };
+        x86_64-darwin = {
+          url = "https://github.com/${pname}/${pname}/releases/download/v${version}/${pname}-${version}-mac_x64.dmg";
```
Comment (SuperSandro2000) [inline pkgs/by-name/db/dbgate/package.nix:23]:
```suggestion
        aarch64-linux = {
          url = "https://github.com/dbgate/dbgate/releases/download/v${version}/dbgate-${version}-linux_arm64.AppImage";
          hash = "sha256-gxojSSk7prhnd9fy56B9H+Cj6COBLc7xPfV8dTvSO0c=";
        };
        x86_64-linux = {
          url = "https://github.com/dbgate/dbgate/releases/download/v${version}/dbgate-${version}-linux_x86_64.AppImage";
          hash = "sha256-/Vfd0R+Mzx1CJKkC7dj99pbuuyh8PJtbYlH3wtwVxSM=";
        };
        x86_64-darwin = {
          url = "https://github.com/dbgate/dbgate/releases/download/v${version}/dbgate-${version}-mac_x64.dmg";
```
substituting pname is an anti pattern

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/db/dbgate/package.nix [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          hash = "sha256-1kC5CNgD3KGR3nd14cBHhYKCThualLKR3CE4KGKh/Hs=";
+        };
+      }
+      .${stdenv.system} or (throw "${pname}-${version}: ${stdenv.system} is unsupported.");
```
Comment (SuperSandro2000) [inline pkgs/by-name/db/dbgate/package.nix:27]:
```suggestion
      .${stdenv.system} or (throw "dbgate: ${stdenv.system} is unsupported.");
```

File: pkgs/by-name/db/dbgate/package.nix (hunk 2 above)
Comment (luftmensch-luftmensch) [reply pkgs/by-name/db/dbgate/package.nix]: Didn't know that. Applied suggestion

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**